lecture2gether-vue/src/components/Toolbar.vue

```diff
@@ -103,7 +103,7 @@ export default class Toolbar extends Vue {
             } else {
                 this.urlIsValid = true;
                 this.$store.dispatch('setUrl', videoMetaData.streamUrl);
-                delete videoMetaData.streamUrl;
+                videoMetaData.streamUrl = '';
```

Comment:
`streamUrl` might not be set on type `VideoMetaData` (as defined in `api.ts`), that is why it was deleted. Setting it to empty string still violates the interface, I think.

Reply:
I have addressed this with e21ffab373d23b4668c9950482ca28bc29c01e74

Reply:
Now I don't really see the improvement. You merge the two interfaces, `VideoMetaData` and `VideoMetaDataWithUrl`, and now the knowledge when which interface is used has vanished from the code, we just have to know when to set/unset the `streamUrl`.
Deleting the `streamUrl` here is actually not strictly necessary, we just chose to do so to avoid duplication of data in the store because the url also *has* to be present in the `videoUrl` field outside of the metadata. So in my opinion we should either delete the `streamUrl` field and keep two interfaces to clearly separate the usages, or we should just keep the `streamUrl` field and have the duplicated data. The downside of the second approach would be that it might be confusing later on why the url is ignored in the meta data.

Reply:
> So in my opinion we should either delete the streamUrl field and keep two interfaces to clearly separate the usages

I don't understand. How does that work when the streamUrl field is the onle difference in the two interfaces?

To be hones, in general I don't quite see how the metadata management works in our code and currently don't really *want* to try to understand it either.

---

The reason I am keen to do something in this part of the code is that the new typescript version does not allow `delete` statements for fields which are defined as non-optional on the target type.
So if simply no having the `delete` statement here is an option, I would be content with that too.

Reply:
Okay, in that case keep the field as optional but add a comment to the interface where you explain that it should only be set for the API and not when it is used later.

Reply:
I really don't understand what you mean by that.
As far as I can see it we commit the streamUrl into the store which then sets it as videoUrl. Is this not a *later usage* which we should not do?
https://github.com/TheNerdful8/Lecture2Gether/blob/521243791771731a2673492105394425e89195db/lecture2gether-vue/src/components/Toolbar.vue#L105-L106
https://github.com/TheNerdful8/Lecture2Gether/blob/521243791771731a2673492105394425e89195db/lecture2gether-vue/src/plugins/store/player.ts#L59-L68